Content.Server/Interaction/InteractionSystem.cs

```diff
@@ -471,47 +469,49 @@ public void DoAttack(EntityUid user, EntityCoordinates coordinates, bool wideAtt
             }
 
             // Verify user has a hand, and find what object they are currently holding in their active hand
-            if (EntityManager.TryGetComponent<HandsComponent?>(user, out var hands))
+            if (TryComp(user, out HandsComponent? hands))
             {
-                if (hands.GetActiveHand?.Owner is {Valid: true} item)
+                var item = hands.GetActiveHand?.Owner;
+
+                if (item != null && !Deleted(item.Value))
                 {
                     if (wideAttack)
                     {
-                        var ev = new WideAttackEvent(item, user, coordinates);
-                        RaiseLocalEvent(item, ev, false);
+                        var ev = new WideAttackEvent(item.Value, user, coordinates);
+                        RaiseLocalEvent(item.Value, ev, false);
 
                         if (ev.Handled)
                         {
-                            _adminLogSystem.Add(LogType.AttackArmedWide, LogImpact.Medium, $"{ToPrettyString(user):user} wide attacked with {ToPrettyString(item):used} at {coordinates}");
+                            _adminLogSystem.Add(LogType.AttackArmedWide, LogImpact.Medium, $"{ToPrettyString(user):user} wide attacked with {ToPrettyString(item.Value):used} at {coordinates}");
                             return;
                         }
                     }
                     else
                     {
-                        var ev = new ClickAttackEvent(item, user, coordinates, targetUid);
-                        RaiseLocalEvent(item, ev, false);
+                        var ev = new ClickAttackEvent(item.Value, user, coordinates, targetUid);
+                        RaiseLocalEvent(item.Value, ev, false);
 
                         if (ev.Handled)
                         {
-                            if (targetUid != default)
+                            if (targetUid != null)
```

Comment:
Is a deleted check needed here?

Reply:
Given this is just for logging, I decided not to add the deleted check. `ToPrettyString` is capable of specifying whether the entity is deleted or not, after all